src/generate.ts

```diff
@@ -180,14 +180,18 @@ function supportDeepObjects(params: OpenAPIV3.ParameterObject[]) {
 export default function generateApi(spec: OpenAPIV3.Document) {
   const aliases: ts.TypeAliasDeclaration[] = [];
 
-  function resolve<T>(obj: T | OpenAPIV3.ReferenceObject) {
-    if (!isReference(obj)) return obj;
-    const ref = obj.$ref;
-    if (!ref.startsWith("#/")) {
-      throw new Error(`External refs are not supported: ${ref}`);
+  function resolve<T>(obj: T | OpenAPIV3.ReferenceObject): T {
+    function resolveAux<T>(obj: T | OpenAPIV3.ReferenceObject): T {
```

Comment:
Why do we need this inner function? Couldn't `resolve` call itself in line 191?

Reply:
You are completely right.